dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -279,36 +279,32 @@ public CultureInfo Culture
     #region Internal Filtering
 
     [Experimental("SKEXP0001")]
-    internal FunctionInvokingContext? OnFunctionInvokingFilter(KernelFunction function, KernelArguments arguments)
+    internal async Task<FunctionInvocationContext?> OnFunctionInvocationAsync(
```

Comment:
Why does this return a nullable context? In what situation does it ever return null?

Reply:
> Why does this return a nullable context? In what situation does it ever return null?

In situation where there are no filters registered in kernel. No filters - no context, just function invocation and result.

Reply:
Even in that case, it's returning non-null:
https://github.com/microsoft/semantic-kernel/pull/5788/files/00175eba09591d8a9bac78d6a3428f8e4d549195#diff-d3103bc841c57ed1f9f7cbd3b3f3be955eca5f85913ae67041a13bd8b66b3daeR287